### ISSUE TITLE
Puppet "heredoc" strings not properly supported

### ISSUE THREAD

**Issue by jearls** [rodjek/puppet-lint#467]
This is a valid Puppet "heredoc" syntax:

```
$foo = @("endOfFoo")
    bar
    ${baz}
    quux
    |-endOfFoo
```

This reports the following problems:
1. It reports a syntax error for `${baz}`, since it doesn't realize this is inside a string value.  A possible workaround is to put the variable interpolation inside ""-quotes in the heredoc, but that only works if the addition of the ""-quotes are transparent to the variable (for example, when it's being used as a shell input):
   
   ```
   $foo = @("endOfFoo")
       bar
       "${baz}"
       quux
       |-endOfFoo
   ```
2. Once the syntax error work-around is employed, the start of the heredoc is marked as a double-quoted string containing no variables.  The double-quotes are a forced part of the heredoc syntax if you want variable interpolation within the heredoc.
3. The above syntax error work-around causes `string containing only a variable` warnings.

**Comment by TheMeier** [comment]
#430
